File: Pacman-Game-OOP-master 2/EASendMail/Samples_ASPNetMvc/VB/WebProject1/WebProject1/Models/DbMailTask.vb
Imports System.ComponentModel.DataAnnotations
Imports EASendMail

Public Class DbMailTask
    <Key, ScaffoldColumn(False)>
    Public Property TaskId As Integer

    <MaxLength(250), Display(Name:="Task Name")>
    Public Property TaskName As String

    <Display(Name:="Server"), MaxLength(128)>
    Public Property Server As String

    <Required>
    Public Property Port As Integer = 25

    <Display(Name:="My server requires user authentication")>
    Public Property IsAuthenticationRequired As Boolean

    <RequiredIf("IsAuthenticationRequired", True, ErrorMessage:="User name is required"), MaxLength(128)>
    Public Property User As String

    <RequiredIf("IsAuthenticationRequired", True, ErrorMessage:="Password is required"), MaxLength(128)>
    Public Property Password As String

    <Display(Name:="TLS/SSL Connection")>
    Public Property IsSslConnection As Boolean

    Public Property Protocol As ServerProtocol = ServerProtocol.SMTP
    Public Property AuthType As SmtpAuthType = SmtpAuthType.AuthAuto

    <Required, Display(Name:="Sender")>
    Public Property Sender As String

    <Display(Name:="Reply-To")>
    Public Property ReplyTo As String

    <Required>
    Public Property Subject As String

    <Display(Name:="Text Body")>
    Public Property TextBody As String

    <Display(Name:="Html Body")>
    Public Property HtmlBody As String

    Public Property Status As DbMailTaskStatus

    <Display(Name:="Total Count")>
    Public Property TotalCount As Integer = 0
    Public Property Succeeded As Integer = 0
    Public Property Failed As Integer = 0

    <Display(Name:="Creation Time")>
    Public Property CreationTime As DateTime = DateTime.Now

    <Display(Name:="Update Time")>
    Public Property LastWriteTime As DateTime = DateTime.Now

    Public Function BuildServer() As SmtpServer
        Dim myServer = New SmtpServer(If(String.IsNullOrWhiteSpace(Server), String.Empty, Server.Trim()))
        myServer.Protocol = Protocol
        myServer.Port = Port
        myServer.ConnectType = If((IsSslConnection), SmtpConnectType.ConnectSSLAuto, SmtpConnectType.ConnectTryTLS)

        If IsAuthenticationRequired Then
            myServer.AuthType = AuthType
            myServer.User = (If(String.IsNullOrWhiteSpace(User), String.Empty, User.Trim()))
            myServer.Password = (If(String.IsNullOrWhiteSpace(Password), String.Empty, Password.Trim()))
        End If

        Return myServer
    End Function

    Public Function BuildMail() As SmtpMail
        Dim mail = New SmtpMail("TryIt")
        mail.From = (If(String.IsNullOrWhiteSpace(Sender), String.Empty, Sender.Trim()))
        mail.Subject = If(String.IsNullOrEmpty(Subject), String.Empty, Subject)

        If Not String.IsNullOrEmpty(ReplyTo) Then
            Dim myReplyTo = New MailAddress(ReplyTo)

            If String.Compare(myReplyTo.Address, mail.From.Address, True) <> 0 Then
                mail.ReplyTo = myReplyTo
            End If
        End If

        If Not String.IsNullOrEmpty(TextBody) Then
            mail.TextBody = TextBody
        End If

        If Not String.IsNullOrEmpty(HtmlBody) Then
            mail.HtmlBody = HtmlBody
        End If

        Return mail
    End Function

    Public Function Copy() As DbMailTask
        Dim mailTask = New DbMailTask()
        mailTask.AuthType = Me.AuthType
        mailTask.CreationTime = Me.CreationTime
        mailTask.HtmlBody = Me.HtmlBody
        mailTask.IsAuthenticationRequired = Me.IsAuthenticationRequired
        mailTask.IsSslConnection = Me.IsSslConnection
        mailTask.LastWriteTime = Me.LastWriteTime
        mailTask.Password = Me.Password
        mailTask.Port = Me.Port
        mailTask.Protocol = Me.Protocol
        mailTask.ReplyTo = Me.ReplyTo
        mailTask.Sender = Me.Sender
        mailTask.Server = Me.Server
        mailTask.Subject = Me.Subject
        mailTask.TaskId = Me.TaskId
        mailTask.TaskName = Me.TaskName
        mailTask.TextBody = Me.TextBody
        mailTask.User = Me.User
        Return mailTask
    End Function
End Class

File: Pacman-Game-OOP-master 2/EASendMail/Samples_ASPNetMvc/C#/WebProject1/WebProject1/Models/DbMailTask.cs
using System.ComponentModel.DataAnnotations;
using EASendMail;
using System;

namespace WebProject1.Models
{
    public class DbMailTask
    {
        [Key, ScaffoldColumn(false)]
        public int TaskId { get; set; }

        [MaxLength(250), Display(Name = "Task Name")]
        public string TaskName { get; set; }

        [Display(Name = "Server"), MaxLength(128)]
        public string Server { get; set; }

        [Required]
        public int Port { get; set; } = 25;

        [Display(Name = "My server requires user authentication")]
        public bool IsAuthenticationRequired { get; set; }

        [RequiredIf("IsAuthenticationRequired", true, ErrorMessage = "User name is required"), MaxLength(128)]
        public string User { get; set; }

        [RequiredIf("IsAuthenticationRequired", true, ErrorMessage = "Password is required"), MaxLength(128)]
        public string Password { get; set; }

        [Display(Name = "TLS/SSL Connection")]
        public bool IsSslConnection { get; set; }

        public ServerProtocol Protocol { get; set; } = ServerProtocol.SMTP;
        public SmtpAuthType AuthType { get; set; } = SmtpAuthType.AuthAuto;

        [Required, Display(Name = "Sender")]
        public string Sender { get; set; }

        [Display(Name = "Reply-To")]
        public string ReplyTo { get; set; }

        [Required]
        public string Subject { get; set; }

        [Display(Name = "Text Body")]
        public string TextBody { get; set; }

        [Display(Name = "Html Body")]
        public string HtmlBody { get; set; }

        public DbMailTaskStatus Status { get; set; }

        [Display(Name = "Total Count")]
        public int TotalCount { get; set; } = 0;

        public int Succeeded { get; set; } = 0;
        public int Failed { get; set; } = 0;

        [Display(Name = "Creation Time")]
        public DateTime CreationTime { get; set; } = DateTime.Now;

        [Display(Name = "Update Time")]
        public DateTime LastWriteTime { get; set; } = DateTime.Now;

        public SmtpServer BuildServer()
        {
            var server = new SmtpServer(string.IsNullOrWhiteSpace(Server) ? string.Empty : Server.Trim());
            server.Protocol = Protocol;

            server.Port = Port;
            server.ConnectType = (IsSslConnection) ?
                SmtpConnectType.ConnectSSLAuto : SmtpConnectType.ConnectTryTLS;

            if (IsAuthenticationRequired)
            {
                server.AuthType = AuthType;
                server.User = (string.IsNullOrWhiteSpace(User) ? string.Empty : User.Trim());
                server.Password = (string.IsNullOrWhiteSpace(Password) ? string.Empty : Password.Trim());
            }

            return server;
        }

        public SmtpMail BuildMail()
        {
            var mail = new SmtpMail("TryIt");
            mail.From = (string.IsNullOrWhiteSpace(Sender) ? string.Empty : Sender.Trim());

            mail.Subject = string.IsNullOrEmpty(Subject) ? string.Empty : Subject;

            if (!string.IsNullOrEmpty(ReplyTo))
            {
                var replyTo = new MailAddress(ReplyTo);
                if (string.Compare(replyTo.Address, mail.From.Address, true) != 0)
                {
                    mail.ReplyTo = replyTo;
                }
            }

            if (!string.IsNullOrEmpty(TextBody))
            {
                mail.TextBody = TextBody;
            }

            if (!string.IsNullOrEmpty(HtmlBody))
            {
                mail.HtmlBody = HtmlBody;
            }

            return mail;
        }

        public DbMailTask Copy()
        {
            var mailTask = new DbMailTask();
            mailTask.AuthType = this.AuthType;
            mailTask.CreationTime = this.CreationTime;
            mailTask.HtmlBody = this.HtmlBody;
            mailTask.IsAuthenticationRequired = this.IsAuthenticationRequired;
            mailTask.IsSslConnection = this.IsSslConnection;
            mailTask.LastWriteTime = this.LastWriteTime;
            mailTask.Password = this.Password;
            mailTask.Port = this.Port;
            mailTask.Protocol = this.Protocol;
            mailTask.ReplyTo = this.ReplyTo;
            mailTask.Sender = this.Sender;
            mailTask.Server = this.Server;
            mailTask.Subject = this.Subject;
            mailTask.TaskId = this.TaskId;
            mailTask.TaskName = this.TaskName;
            mailTask.TextBody = this.TextBody;
            mailTask.User = this.User;

            return mailTask;
        }
    }
}